homeassistant/components/guardian/switch.py

```diff
@@ -112,14 +102,15 @@ async def _async_continue_entity_setup(self):
     @callback
     def _async_update_from_latest_data(self) -> None:
         """Update the entity."""
-        self._is_on = self.coordinators[API_VALVE_STATUS].data["state"] in (
+        self._attr_available = self.coordinators[API_VALVE_STATUS].last_update_success
```

Comment:
It seems like this entity is using a single coordinator.

In that case, this class should extend from the `CoordinatorEntity` and pass in the coordinator in the parent constructor.

That removes the need to manually manage the availability and allows for accessing the coordinator directly in the rest of the entity as an instance variable (`self.coordinator`).

This pattern seems similar in other entities.

Reply:
Not quite. This entity inherits from `ValveControllerEntity`, which needs to access at least two coordinators (the one assigned to this entity's data _and_ the one assigned to diagnostics data).

Reply:
Right! Thanks for clearing that up